File: MockInterview.Backend/MockInterview.BLL/Models/DTOs/TalentDto.cs
using MockInterview.BLL.Models.Common;

namespace MockInterview.BLL.Models.DTOs;

/// <summary>
/// Represents talented person
/// </summary>
public class TalentDto : IEntityDto
{
    public long Id { get; set; }
    
    /// <summary>
    /// Current level
    /// </summary>
    public string Level { get; set; } = null!;

    /// <summary>
    /// Earned experience
    /// </summary>
    public string Experience { get; set; } = null!;

    /// <summary>
    /// Worked projects
    /// </summary>
    public string Projects { get; set; } = null!;
}

File: MockInterview.Frontend/src/types/talentDto.ts
import {IMappable} from "@/types/common";

export interface ITalentDto {
    id: number;
    level: string;
    experience: string;
    projects: string;
}

export class TalentDto implements IMappable<ITalentDto> {
    id!: number;
    level!: string;
    experience!: string;
    projects!: string;

    mapFrom(data: ITalentDto) {
        this.id = data.id;
        this.level = data.level;
        this.experience = data.experience;
        this.projects = data.projects;
    }
}